restful.go

```diff
@@ -199,7 +203,31 @@ func postRestfulQueryHelper(
 
 		var resultURL string
 		isSessionRenewed := false
+		noResult, _ := isAsyncMode(ctx)
+
+		// if asynchronous query in progress, kick off retrieval but return object
+		if respd.Code == queryInProgressAsyncCode && noResult {
+			// placeholder object to return to user while retrieving results
+			rows := new(snowflakeRows)
+			res := new(snowflakeResult)
+			switch resType, _ := ctx.Value(resultType).(string); resType {
+			case "exec":
+				res.queryID = respd.Data.QueryID
+				res.status = QueryStatusInProgress
+				res.statusChannel = make(chan queryStatus)
+				respd.Data.AsyncResult = res
+			case "query":
+				rows.queryID = respd.Data.QueryID
+				rows.status = QueryStatusInProgress
+				rows.statusChannel = make(chan queryStatus)
+				respd.Data.AsyncRows = rows
+			default:
```

Comment:
Should this default case fail query execution? The queryID, status, and statusChannel not being set feels like undefined behavior once the goroutine is kicked off below.

Reply:
Good point. I'll short circuit this and return the returned &respd object before the goroutine is kicked off.